src/Upload.tsx

```diff
@@ -0,0 +1,129 @@
+import React, {
+    DetailedHTMLProps,
+    InputHTMLAttributes,
+    ReactNode,
+    forwardRef,
+    memo,
+    useId
+} from "react";
+import { createComponentI18nApi } from "./i18n";
+import { symToStr } from "tsafe/symToStr";
+import { cx } from "./tools/cx";
+import { fr } from "./fr";
+import { Equals, assert } from "tsafe";
+
+export type UploadProps = {
+    className?: string;
+    /** @default false */
+    disabled?: boolean;
+    hint?: string;
+    /** @default false */
+    multiple?: boolean;
+    /** @default "default" */
+    state?: "success" | "error" | "default";
+    /** The message won't be displayed if state is "default" */
+    stateRelatedMessage?: ReactNode;
+    /** Props forwarded to the underlying <input /> element */
+    nativeInputProps?: DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>;
+};
+
+export const Upload = memo(
+    forwardRef<HTMLDivElement, UploadProps>((props, ref) => {
+        const { t } = useTranslation();
+        const {
+            className,
+            disabled = false,
+            hint = t("hint"),
+            multiple = false,
+            state = "default",
+            stateRelatedMessage,
+            nativeInputProps = {}
+        } = props;
+
+        const inputId = (function useClosure() {
+            const id = useId();
+
+            return nativeInputProps.id ?? `input-${id}`;
+        })();
+
+        const messageId = `${inputId}-desc-error`;
+        return (
+            <div
+                className={cx(
+                    fr.cx(
+                        "fr-upload-group",
+                        disabled && "fr-input-group--disabled",
+                        (() => {
+                            switch (state) {
+                                case "error":
+                                    return "fr-input-group--error";
+                                case "success":
+                                    return "fr-input-group--valid";
+                                case "default":
+                                    return undefined;
+                            }
+                            assert<Equals<typeof state, never>>(false);
+                        })()
+                    ),
+                    className
+                )}
+                ref={ref}
+            >
+                <label className="fr-label" aria-disabled={disabled} htmlFor={inputId}>
```

Comment:
```suggestion
                <label className={fr.cx("fr-label")} aria-disabled={disabled} htmlFor={inputId}>
```
Please type check **every** fr-classes.  
If a class get removed from the DSFR we want to know about it and stop using it.